dashboard/app/views/levels/_free_response.html.haml

```diff
@@ -1,16 +1,33 @@
-:javascript
-  function getResult() {
-    return {
-      response: $('.response').val(),
-      result: true
-    };
-  }
+- level ||= @level
+- last_attempt ||= @last_attempt
+- index ||= nil
+- number ||= nil
+
+- if index
+  :javascript
+    window['level_#{index}'] = {
```

Comment:
Brent suggested storing the results on a single object in a new `levelGroup` namespace, instead of having multiple global vars.  I put together an example of what that could look like here: https://github.com/code-dot-org/code-dot-org/compare/free-response-level-group...level-group-results.